File: app/src/androidTest/java/com/brunoaybar/unofficialupc/data/models/UserTest.java
package com.brunoaybar.unofficialupc.data.models;

import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Created by brunoaybar on 10/01/2017.
 */

public class UserTest {
    @Test
    public void hasInvalidSession_NoToken_NoCode() throws Exception {
        User user = new User(null,null,null);
        assertEquals(false,user.hasValidSession());
    }
    @Test
    public void hasInvalidSession_NoToken() throws Exception {
        User user = new User(null,"test",null);
        assertEquals(false,user.hasValidSession());
    }

    @Test
    public void hasInvalidSession_NoCode() throws Exception {
        User user = new User("test",null,null);
        assertEquals(false,user.hasValidSession());
    }
    @Test
    public void hasInvalidSession_EmptyToken_EmptyCode() throws Exception {
        User user = new User("","",null);
        assertEquals(false,user.hasValidSession());
    }
    @Test
    public void hasInvalidSession_EmptyToken() throws Exception {
        User user = new User("","test",null);
        assertEquals(false,user.hasValidSession());
    }

    @Test
    public void hasInvalidSession_EmptyCode() throws Exception {
        User user = new User("test","",null);
        assertEquals(false,user.hasValidSession());
    }

    @Test
    public void hasValidSession() throws Exception {
        User user = new User("test","test",null);
        assertEquals(true,user.hasValidSession());
    }


    @Test
    public void hasInvalidCredentials_NoCode_NoPass() throws Exception {
        User user = new User(null,null,null);
        assertEquals(false,user.hasValidCredentials());
    }
    @Test
    public void hasInvalidCredentials_NoCode() throws Exception {
        User user = new User(null,null,"test");
        assertEquals(false,user.hasValidCredentials());
    }
    @Test
    public void hasInvalidCredentials_NoPass() throws Exception {
        User user = new User(null,"test",null);
        assertEquals(false,user.hasValidCredentials());
    }
    @Test
    public void hasInvalidCredentials_EmptyCode_EmptyPass() throws Exception {
        User user = new User("","","");
        assertEquals(false,user.hasValidCredentials());
    }
    @Test
    public void hasInvalidCredentials_EmptyCode() throws Exception {
        User user = new User("","","test");
        assertEquals(false,user.hasValidCredentials());
    }
    @Test
    public void hasInvalidCredentials_EmptyPass() throws Exception {
        User user = new User("","test","");
        assertEquals(false,user.hasValidCredentials());
    }
    @Test
    public void hasValidCrendital() throws Exception {
        User user = new User(null,"user","password");
        assertEquals(true,user.hasValidCredentials());
    }

}

File: app/src/test/kotlin/com/brunoaybar/unofficialupc/data/models/UserTest.kt
package com.brunoaybar.unofficialupc.data.models

import org.junit.Test
import kotlin.test.assertFalse

class UserTest{

    private val TEST_TOKEN = "test"
    private val TEST_CODE = "U201313295"
    private val TEST_PASS = "test"

    @Test
    fun invalidSessionWhen_NoToken_NoCode(){
        val user = User(null,null,null)
        assertFalse { user.hasValidSession() }
    }
    @Test
    fun invalidSessionWhen_HasToken_NoCode(){
        val user = User(TEST_TOKEN,null,null)
        assertFalse { user.hasValidSession() }
    }
    @Test
    fun invalidSessionWhen_HasCode_NoToken(){
        val user = User(null,TEST_CODE,null)
        assertFalse { user.hasValidSession() }
    }

    @Test
    fun validSessionWhen_HasToken_HasCode(){
        val user = User(TEST_TOKEN,TEST_CODE,null)
        assert(user.hasValidSession())
    }


    @Test
    fun invalidCredentialsWhen_NoCode_NoPass(){
        val user = User(null,null,null)
        assertFalse { user.hasValidCredentials() }
    }
    @Test
    fun invalidCredentialsWhen_HasCode_NoPass(){
        val user = User(null,TEST_CODE,null)
        assertFalse { user.hasValidCredentials() }
    }
    @Test
    fun invalidCredentialsWhen_NoCode_HasPass(){
        val user = User(null,null,TEST_PASS)
        assertFalse { user.hasValidCredentials() }
    }
    @Test
    fun validCredentialsWhen_HasCode_HasPass(){
        val user = User(null,TEST_CODE,TEST_PASS)
        assert(user.hasValidCredentials())
    }

}

